processing/src/main/java/io/druid/query/groupby/epinephelinae/ByteBufferIntList.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.groupby.epinephelinae;
+
+import com.google.common.primitives.Ints;
+import com.metamx.common.IAE;
+
+import java.nio.ByteBuffer;
+
+public class ByteBufferIntList
```

Comment:
Hm, I'm leaning towards keeping this as-is for now, since it's only being used for a narrow use case by the BufferGrouper right now and the AbstractList has quite a few methods to implement, I feel like it would be too large for this PR, without being necessary for the push down changes